src/RadixDlt.NetworkGateway.PostgresIntegration/DataAggregatorBuilderExtensions.cs

```diff
@@ -91,7 +92,9 @@ public static DataAggregatorBuilder AddPostgresPersistenceCore(this DataAggregat
             .AddSingleton<ITopOfLedgerProvider, TopOfLedgerProvider>()
             .AddSingleton<INetworkConfigurationProvider, NetworkConfigurationProvider>()
             .AddSingleton<IPendingTransactionResubmissionService, PendingTransactionResubmissionService>()
-            .AddSingleton<IPendingTransactionPrunerService, PendingTransactionPrunerService>();
+            .AddSingleton<IPendingTransactionPrunerService, PendingTransactionPrunerService>()
+            .AddSingleton<IDapperWrapper, DapperWrapper>()
+            .AddSingleton<MetricsInterceptor>();
```

Comment:
I think it would be safer to make the interceptor and dapper wrapper `.AddScoped` (GW API + DA) in order to a) follow db context scope conventions, b) make DapperWrapper dependant on `IOptions[snapshot]` rather `IOptionsMonitor`.

Reply:
I don't see easy option to keep MetricsInterceptor as scoped and register that interceptor for db context.